Sources/Documentation/XCStringsToolPlugin.docc/CONTRIBUTING.md

```diff
@@ -0,0 +1,49 @@
+# Contributing
+
+Instructions on how to contribute to the project.
+
+## Overview
+
+Contributions to XCStrings Tool are welcome!
+
+- For ideas and questions: [Visit the Discussions](https://github.com/liamnichols/xcstrings-tool/discussions)
+
+- For bugs: [Open an Issue](https://github.com/liamnichols/xcstrings-tool/issues/choose)
+
+- For contributions: [Open a Pull Request](https://github.com/liamnichols/xcstrings-tool/compare)
+
+## Testing your contributions
+
+To build and run all test cases open XCStrings Tool with Xcode and hit ⌘U (command + U).
+
+### Snapshot tests in XCStrings Tool
+
+The majority of tests for XCStrings Tool are implemented using snapshot tests. Snapshot tests compare a "snapshot" of the output generated by a previous version of the software with the output generated by a subsequent version. The test is passed if the outputs are the same.  
+
+Snapshot testing is handled by the [GenerateTests](https://github.com/liamnichols/xcstrings-tool/blob/main/Tests/XCStringsToolTests/GenerateTests.swift) class. When the snapshot tests are run, XCStrings Tool generates Swift code using the .xcstrings files in the [Fixtures](https://github.com/liamnichols/xcstrings-tool/tree/main/Tests/XCStringsToolTests/__Fixtures__) directory. These files are then compared to the snapshots in the [GenerateTests](https://github.com/liamnichols/xcstrings-tool/tree/main/Tests/XCStringsToolTests/__Snapshots__/GenerateTests) directory.
+
+### Dealing with failed snapshot tests
+
+If there is a discrepancy between a generated Swift file and its corresponding snapshot, the relevant snapshot test will fail. This means that either:
+
+* A bug has been introduced; or
+
+* The changes that have been made are expected to generate a new Swift file. In this case the relevant snapshot file in [GenerateTests](https://github.com/liamnichols/xcstrings-tool/tree/main/Tests/XCStringsToolTests/__Snapshots__/GenerateTests) must be updated.
```

Comment:
If the snapshot test generates an expected diff, the test will also fail, but this is because we want to re-record the snapshot.

In this case, you need to set `record` to `true`:

https://github.com/liamnichols/xcstrings-tool/blob/0c77774313bb148f726b1d31b016a4e739c85f6e/Tests/XCStringsToolTests/GenerateTests.swift#L55

Would you also be able to mention this? 🙏